#215 add target attribute explicitAccessUrl

sfrevel opened the pull request:
see https://github.com/promregator/promregator/issues/215

closes #215

checketts commented:
Can you please add an explanation of why this is needed and perhaps the supporting code that leverages it?

'explicitAccessUrl' also isn't clear. is a better name 'routeOverride' (I'm assuming it ignores the `preferredRouteRegex`)?

eaglerainbow reviewed (commented):
Mainly cosmetics.
Unit tests and documentation text are still missing, though.

eaglerainbow commented:
> Can you please add an explanation of why this is needed and perhaps the supporting code that leverages it?
> 
> 'explicitAccessUrl' also isn't clear. is a better name 'routeOverride' (I'm assuming it ignores the `preferredRouteRegex`)?

BTW: I don't think that it's a route overwrite. How it is implemented, it's a `hostOverride`. 
@sfrevel What's the expectation here? What is needed? Do you also need to be able to mangle with the path in certain cases?

checketts commented on src/main/java/org/cloudfoundry/promregator/scanner/ReactiveAppInstanceScanner.java:
```diff
@@ -16,6 +16,7 @@
 import org.cloudfoundry.client.v2.spaces.SpaceApplicationSummary;
 import org.cloudfoundry.client.v2.spaces.SpaceResource;
 import org.cloudfoundry.client.v2.routes.Route;
+import org.apache.logging.log4j.util.Strings;
```
Log4j is on the outs at the moment. 😉

eaglerainbow replied:
Well, that itself isn't the main reason 😁 
And BTW those folks IMHO do the best that they can right now. Instead of slamming our doors into their face, we should be thankful for them to be transparent...

checketts replied:
I agree. Sorry my joke wasn't intended to be insensetive.

eaglerainbow replied:
no worries. Didn't read it like this; but these poor guys need our support - even if it is only mentally.

sfrevel replied:
Ok I used this, because the library was already in the dependencies. And I think log4j was updated in the meanwhile...

checketts commented:
I overlooked the changes in `ReactiveAppInstanceScanner.java ` somehow when I looked at this last time, just saw `Target` now it is more complete. 

I proposed it `routeOverride` just to stay parallel with `preferredRouteRegex`.  I don' think it is a `hostOverride` since it coves the entire URL (protocol, host, port, path)

I'm not opposed to `explictAccessUrl` but `access` doesn't seem to be a term we've used before. We use the `preferredRouteRegex` + `path`, so perhaps `overrideRouteAndPath` is more accurate, but confusing. URL might be the best name if the value is a complete url.

sfrevel commented:
Sorry for the late response. We where putting all things together in the meanwhile - and the small changes with the property are doing really well. So I renamed the property to the last suggestion 'overrideRouteAndPath' and added the other missing thing like the test and doc of the property.

eaglerainbow commented:
BTW: Please don't worry about the CI/pr-merge voter complaining right now. I'll take care of that as soon as the reviews are through (don't want to have unnecessary work 😁 )
